### ISSUE TITLE
weird format of queries when "Preserve original" is selected

### ISSUE THREAD

**Issue by krokyk** [Trivadis/plsql-formatter-settings#18]
The table name is not on the same line as update/insert statement but below
![image](https://user-images.githubusercontent.com/15148711/89034997-66886480-d33a-11ea-96c6-bf2f36b352c8.png)
![image](https://user-images.githubusercontent.com/15148711/89035096-9c2d4d80-d33a-11ea-8f64-1937fa6ad998.png)

**Comment by PhilippSalvisberg** [comment]
Can you please provide the code as text as well? Thank you.

**Comment by krokyk** [comment]
Oh sorry, sure:
```sql
    FUNCTION merge_config_entry (
        p_namespace         configuration.namespace%TYPE,
        p_key               configuration.key%TYPE,
        p_scope             configuration.scope%TYPE,
        p_value             configuration.value%TYPE,
        p_visible           configuration.visible%TYPE,
        p_register_zeichen  configuration.register_zeichen%TYPE,
        p_is_blob           BOOLEAN
    ) RETURN configuration.id%TYPE IS
        l_type  VARCHAR2(100) := get_config_entry_type(p_key, p_is_blob);
        l_id    configuration.id%TYPE := get_config_entry_id(p_namespace, p_key, p_scope);
    BEGIN
        IF l_id IS NULL THEN
            l_id := insert_config_entry(p_namespace, p_key, p_scope, p_value, p_visible, p_register_zeichen, p_is_blob);
        ELSE
            BEGIN
                print('Updating existing %s entry: %s:%s:%s:%s:%s:%s:%s... ', l_type, son(l_id), son(p_namespace), son(p_key), son(p_scope), son(p_value), son(p_visible), son(p_register_zeichen));

                UPDATE
                configuration
                   SET value = p_value,
                       visible = p_visible,
                       previous_in_order = 'deprecated',
                       register_zeichen = p_register_zeichen
                 WHERE id = l_id;

                dbms_output.put_line('OK');
            EXCEPTION
                WHEN OTHERS THEN
                    dbms_output.put_line('FAILED');
                    RAISE;
            END;
        END IF;

        RETURN l_id;
    END merge_config_entry;
```

**Comment by PhilippSalvisberg** [comment]
Strange. The following code block works:

```sql
BEGIN
    print('Updating existing %s entry: %s:%s:%s:%s:%s:%s:%s... ', l_type, son(l_id), son(p_namespace), son(p_key), son(p_scope),
    son(p_value), son(p_visible), son(p_register_zeichen));

    UPDATE configuration
       SET value = p_value,
           visible = p_visible,
           previous_in_order = 'deprecated',
           register_zeichen = p_register_zeichen
     WHERE id = l_id;

    dbms_output.put_line('OK');
END;
```

I've seen that in other cases. I wanted to switch back to "Preserve Original", but some of my test cases failed with "unexpected line breaks".

Needs more analysis...

**Comment by krokyk** [comment]
Yes, noticed in anonymous block or plain sql it works, just not inside the body of a function/proc

**Comment by PhilippSalvisberg** [comment]
You can change line size to 400 and it will work ;-)
... I'll look at it later.

**Comment by PhilippSalvisberg** [comment]
see question in SQLDev forum: https://community.oracle.com/message/15665616#15665616 
I assume it is a SQLDev bug.

**Comment by PhilippSalvisberg** [comment]
We have to wait until SQLDev 20.3 becomes available to check, if it works then. Can't be fixed through Arbori. 

However, it is possible to change the Arbori program to reduce the risk of long lines. But that's not really a solution IMO.

**Comment by krokyk** [comment]
Thanks for bringing it up with SQLDeveloper dev team. OK, we'll see.

**Comment by PhilippSalvisberg** [comment]
The fix for #19 affected this issue as well. The result looks like this now:

```sql
    FUNCTION merge_config_entry (
        p_namespace         configuration.namespace%TYPE,
        p_key               configuration.key%TYPE,
        p_scope             configuration.scope%TYPE,
        p_value             configuration.value%TYPE,
        p_visible           configuration.visible%TYPE,
        p_register_zeichen  configuration.register_zeichen%TYPE,
        p_is_blob           BOOLEAN
    ) RETURN configuration.id%TYPE IS
        l_type  VARCHAR2(100) := get_config_entry_type(p_key, p_is_blob);
        l_id    configuration.id%TYPE := get_config_entry_id(p_namespace, p_key, p_scope);
    BEGIN
        IF l_id IS NULL THEN
            l_id := insert_config_entry(
                p_namespace,
                p_key,
                p_scope,
                p_value,
                p_visible,
                p_register_zeichen,
                p_is_blob
            );
        ELSE
            BEGIN
                print(
                    'Updating existing %s entry: %s:%s:%s:%s:%s:%s:%s... ',
                    l_type,
                    son(l_id),
                    son(p_namespace),
                    son(p_key),
                    son(p_scope),
                    son(p_value),
                    son(p_visible),
                    son(p_register_zeichen)
                );

                UPDATE configuration
                   SET value = p_value,
                       visible = p_visible,
                       previous_in_order = 'deprecated',
                       register_zeichen = p_register_zeichen
                 WHERE id = l_id;

                dbms_output.put_line('OK');
            EXCEPTION
                WHEN OTHERS THEN
                    dbms_output.put_line('FAILED');
                    RAISE;
            END;
        END IF;

        RETURN l_id;
    END merge_config_entry;
```

However, the issue regarding the scope of long lines need to be fixed in the SQLDev code base. Therefore I close this issue.
